presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockEncoding.java

```diff
@@ -74,8 +63,11 @@ public void writeBlock(SliceOutput sliceOutput, Block block)
 
         int entriesStartOffset = offsets[offsetBase];
         int entriesEndOffset = offsets[offsetBase + positionCount];
-        keyBlockEncoding.writeBlock(sliceOutput, mapBlock.getKeys().getRegion(entriesStartOffset, entriesEndOffset - entriesStartOffset));
-        valueBlockEncoding.writeBlock(sliceOutput, mapBlock.getValues().getRegion(entriesStartOffset, entriesEndOffset - entriesStartOffset));
+
+        TypeSerde.writeType(sliceOutput, mapBlock.keyType);
+
+        blockEncodingSerde.writeBlock(sliceOutput, mapBlock.getKeys().getRegion(entriesStartOffset, entriesEndOffset - entriesStartOffset));
```

Comment:
maybe add a helper to mapBlock to get the key block... actually, maybe rename `getKeys` to `getRawKeysBlock`, and make this the default behavior

Reply:
Done renaming. I will try out introducing the helper separately.